src/compiler/core.ts

```diff
@@ -2154,4 +2167,74 @@ namespace ts {
     export function fill<T>(length: number, cb: (index: number) => T): T[] {
         return new Array(length).fill(0).map((_, i) => cb(i));
     }
+
+    /**
+     * A list of sorted and unique values. Optimized for best performance when items are added
+     * in the sort order.
+     */
+    export class SortedUniqueList<T> implements Push<T> {

```

Comment:
Instead of including this, please create separate PR since the PR doesn't depened on this change and we can weigh in benefits of this and have a separate commit if we want to revert it any time.